packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -108,6 +108,13 @@ export class LambdaInvoke extends sfn.TaskStateBase {
       );
     }
 
+    if (this.props.payloadResponseOnly &&
+      this.props.payload?.value !== undefined && typeof(this.props.payload?.value) !== 'object') {
+      throw Error(
+        `Invalid LambdaInvoke payload value for 'payloadResponseOnly' invocation. Expected 'object' or 'undefined', got '${typeof(this.props.payload?.value)}'.`,
+      );
+    }
```

Comment:
let's make sure we add unit tests for the control flow here. I did see one that mentioned invalid payload, but we should have specific tests for `object`, `undefined`, and `invalid` .

nit: since payload response only validation is getting a few conditions added to it, let's extract this logic out of the constructor to keep the constructor short.

Reply:
Unit tests for object and invalid cases already provided - I will write an additional one for `undefined`.